### PR TITLE
[WIP] Track events 

### LINKED ISSUE #287: Event tracking with Google Analytics
Event is not a main conversion (goal) but still an important action on the website 

- pop up load
- pop up close

Which ultimately gives more data about the conversion of the pop-up like this 

(Ebook downloads through pop-up / sessions with pop-up shown) * 100% = realistic pop-up conversion rate

Or even

(Ebook downloads through pop-up / users who see the pop-up) * 100% 

- click a button (LS page for example) 

Categories: 
Pop-ups, buttons, maybe more

Started working on this this afternoon.

### PR DESCRIPTION
**Pull request by RonaldDijkstra**
Fixes #287 

- [x] load and close popup
- [x] embedded videos
- [x] some buttons

### CODE REVIEW

File: source/javascripts/_popup.js [hunk 1]
```diff
@@ -50,6 +50,7 @@ Defacto.popup = {
           $('.popup').addClass('popup-show');
           Defacto.popup.popupsShown = true;
         }, Defacto.popup.showDelay);
+        ga('send', 'event', 'popup', 'load', $(location).attr('pathname'));
```
Comment (sn3p) [inline source/javascripts/_popup.js]: 1. You probably want to trigger the event inside the `setTimeout` callback.  
Right now it's triggered multiple times before the popup is shown.
2. `$(location).attr('pathname')` can be replaced with `window.location.pathname`.

**TIP:** To read values or test timing you can write `console.log(window.location.pathname)` (or something similar). Output will be show in the browser console.

File: source/localizable/_popup-ebook.erb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <div class="popup" id="popup-ebook">
-  <a href="#" class="close">
+  <a href="#" class="close" onClick="ga('send', 'event', 'popup', 'close', '<%= current_page.url %>');">
```
Comment (sn3p) [inline source/localizable/_popup-ebook.erb]: I would move this to [_popup.js](https://github.com/DefactoSoftware/website/blob/f589d5206b850a251ce6fa6fb2b57e3310073a17/source/javascripts/_popup.js#L58-L62):

```js
$(document).on('click', '.popup .close', function (event) {
  event.preventDefault();
  var $popup = $(this).closest('.popup');
  Defacto.popup.hide($popup, Defacto.popup.cookieExpiresOnClose);
  ga('send', 'event', 'popup', 'close', window.location.pathname);
});
```

**Comment by sn3p** [comment]
Almost there 👍 (see inline comments)

**Comment by RonaldDijkstra** [comment]
Used 2 open source js plugins to make this work.

**Comment by RonaldDijkstra** [comment]
Forget to remove a line :-)

File: source/javascripts/_events.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+$(function () {
+  $('[data-event]').on('click', function () {
+    ga('send', 'event', 'button', 'click', $(this).data('event'));
+    console.log($(this).data('event'));
```
Comment (fatboypunk) [inline source/javascripts/_events.js]: this one? :D

**Comment by fatboypunk** [comment]
Nice going here I guess the following screenshot is from this PR?

File: source/javascripts/_events.js (hunk 3 above)
Comment (RonaldDijkstra) [inline source/javascripts/_events.js]: Yep that one. :-)

**Comment by RonaldDijkstra** [comment]
Yep

**Comment by fatboypunk** [comment]
Really nice 👏

**Comment by RonaldDijkstra** [comment]
I hope it's ready now. 

More screenshots of live tab in Google Analytics: 

live-events-01
live-events-02

Thanks @sn3p for your help with the jquery stuff.

File: source/blog/nl/2016-06-30-zorg-dat-leren-voor-je-werkt.html.markdown [hunk 4]
```diff
@@ -11,16 +11,16 @@ Een intrigerende slogan die de MUMC+ Academie gebruikt als leidraad voor haar le
...
 
 In dit fragment licht Michel van Zandvoort toe hoe het werkplekleren en het leren reflecteren  binnen MUMC vorm krijgt. Ze noemen het klinisch redeneren, een term in de taal van de werkvloer.
 
-<iframe width="560" height="315" src="https://www.youtube.com/embed/Bfe9VqZseW8" frameborder="0" allowfullscreen></iframe>
+<iframe src="https://www.youtube.com/embed/Bfe9VqZseW8?rel=0" frameborder="0" allowfullscreen></iframe>
 
 Een passende leeromgeving is ook een zeer belangrijke voorwaarde om de visie te vertalen naar de praktijk. Deze leeromgeving ondersteunt dan het verwerven van competenties, het aantoonbaar bekwaam zijn en de leerondersteuning op de werkplek. En zo’n leeromgeving is in de visie van MUMC meer dan een LMS alleen. In dit fragment worden de keuzes die het MUMC heeft gemaakt toegelicht.
 
-<iframe width="560" height="315" src="https://www.youtube.com/embed/Ujuq0V31fn0" frameborder="0" allowfullscreen></iframe>
+<iframe src="https://www.youtube.com/embed/Ujuq0V31fn0?rel=0" frameborder="0" allowfullscreen></iframe>
```
Comment (RonaldDijkstra) [inline source/blog/nl/2016-06-30-zorg-dat-leren-voor-je-werkt.html.markdown]: Rel=0 doesn't give related video at the end of the video.

File: source/localizable/elearning/_elearning.de.md.erb [hunk 5]
```diff
@@ -1,6 +1,6 @@
 Defacto ist ein erfahrender E-Learning Partner. Von den allerersten Webinaren in den 90zigern bis hin zu den vielfältigen Möglichkeiten von heute. Dies bedeutet für Ihre Organisation, dass wir nahtlos an Ihre Fragen und Bedürfnisse anschließen können, ob es sich um ein einfaches Modul oder ein anspruchsvolles Medienprodukt handelt. Mit Defacto als sowohl didaktisch und technisch geschultem Partner wird die Verwendung von E-Learning in Ihrer Organisation unwiderruflich zum Erfolg.
 
-<iframe width="560" height="315" src="https://www.youtube.com/embed/4meKUgypYmc?rel=0&amp;showinfo=0&amp;cc_lang_pref=de&amp;cc_load_policy=1" frameborder="0" allowfullscreen></iframe>
+<iframe src="https://www.youtube.com/embed/4meKUgypYmc?rel=0&amp;showinfo=0&amp;cc_lang_pref=de&amp;cc_load_policy=1" frameborder="0" allowfullscreen></iframe>
```
Comment (RonaldDijkstra) [inline source/localizable/elearning/_elearning.de.md.erb]: Removed height and width and let css take care of it as suggested earlier.

**Comment by sn3p** [comment]
Nice @RonaldDijkstra 👍 

Lets put vendor assets in a vendor dir:

```
javascripts/vendor/_youtube.ga.js
javascripts/vendor/_vimeo.ga.js
```

**Pull request merged**